web/index.js

```diff
@@ -80,17 +79,20 @@ map.on('load', function() {
         }).setLngLat(e.lngLat).setHTML(html).addTo(map);
     };
 
-    map.on('click', 'stop-label-intercity', on_stop);
-    map.on('click', 'stop-label-urban', on_stop);
+    map.on('click', 'stops-intercity-icon', on_stop);
+    map.on('click', 'stops-urban-icon', on_stop);
 });
 
 
 function filter_on_one_route(route) {
     //route_id = -7773405
-    var route_id = route['rel_osm_id'];
-    map.setFilter('transport_ways_filtered_outline', ["==", "rel_osm_id", route_id]);
-    map.setFilter('transport_ways_filtered', ["==", "rel_osm_id", route_id]);
-    map.setFilter('transport_points_filtered', ["==", "rel_osm_id", route_id]);
+    var ways_ids = route['ways_ids'];
+    var positions_ids = route['positions_ids'];
+    map.setFilter('routes_ways_filtered_outline', ["in", "id"].concat(ways_ids));
+    map.setFilter('routes_ways_filtered', ["in", "id"].concat(ways_ids));
+    map.setFilter('routes_points_filtered', ["in", "id"].concat(positions_ids));
+    map.setPaintProperty('routes_ways_filtered', "line-color", route['rel_colour'] || 'grey')
+    map.setPaintProperty('routes_points_filtered', "circle-color", route['rel_colour'] || 'grey')
```

Comment:
* I pick the `gray` because in the popup is already the default colour when the route does not have one.
* I think we should change the shield default colour. For the shield we need to provide the default colour directly from the SQL.

Reply:
Done, swithed to gray.